src/main/java/com/github/mc1arke/sonarqube/plugin/CommunityBranchPlugin.java

```diff
@@ -67,12 +73,106 @@ public void load(CoreExtension.Context context) {
+                        .category(PULL_REQUEST_CATEGORY_LABEL)
+                        .subCategory(GENERAL)
+                        .onlyOnQualifiers(Qualifiers.PROJECT)
+                        .name("Provider")
+                        .type(PropertyType.SINGLE_SELECT_LIST)
+                        .options("Github", "BitbucketServer", "BitbucketCloud").build(),
+
+                PropertyDefinition.builder(PullRequestBuildStatusDecorator.PULL_REQUEST_COMMENT_SUMMARY_ENABLED)
+                        .category(PULL_REQUEST_CATEGORY_LABEL)
+                        .subCategory(GENERAL)
+                        .onlyOnQualifiers(Qualifiers.PROJECT)
+                        .name("Enable summary comment")
+                        .description("This enables the summary comment (if implemented).")
+                        .type(PropertyType.BOOLEAN)
+                        .defaultValue("true")
+                        .build(),
+
+                PropertyDefinition.builder(PullRequestBuildStatusDecorator.PULL_REQUEST_FILE_COMMENT_ENABLED)
+                        .category(PULL_REQUEST_CATEGORY_LABEL)
+                        .subCategory(GENERAL)
+                        .onlyOnQualifiers(Qualifiers.PROJECT)
+                        .name("Enable file comment")
+                        .description("This enables commenting (if implemented).")
+                        .type(PropertyType.BOOLEAN)
+                        .defaultValue("true")
+                        .build(),
+
+                PropertyDefinition.builder(PullRequestBuildStatusDecorator.PULL_REQUEST_DELETE_COMMENTS_ENABLED)
+                        .category(PULL_REQUEST_CATEGORY_LABEL)
+                        .subCategory(GENERAL)
+                        .onlyOnQualifiers(Qualifiers.PROJECT)
+                        .name("Enable deleting comments")
+                        .description("This cleans up the comments from previous runs (if implemented).")
+                        .type(PropertyType.BOOLEAN)
+                        .defaultValue("false")
+                        .build(),
+
+                PropertyDefinition.builder(BitbucketServerPullRequestDecorator.PULL_REQUEST_BITBUCKET_URL)
+                        .category(PULL_REQUEST_CATEGORY_LABEL)
+                        .subCategory(BITBUCKET_INTEGRATION_SUBCATEGORY_LABEL)
+                        .onQualifiers(Qualifiers.PROJECT)
+                        .name("URL for Bitbucket (Server or Cloud) instance")
+                        .description("Example: http://bitbucket.local")
+                        .type(PropertyType.STRING)
+                        .build(),
+
+                PropertyDefinition.builder(BitbucketServerPullRequestDecorator.PULL_REQUEST_BITBUCKET_TOKEN)
+                        .category(PULL_REQUEST_CATEGORY_LABEL)
+                        .subCategory(BITBUCKET_INTEGRATION_SUBCATEGORY_LABEL)
+                        .onQualifiers(Qualifiers.PROJECT)
```

Comment:
Should this be `Qualifier.PROJECT` or `Qualifier.APP`? Is `PROJECT` then the token needs set in every project in SonarQube, if `APP` then it should be set globally across all projects?

Reply:
The way I configured it now is
onQualifiers(Qualifiers.PROJECT) this allows us to set a default value for some properties in the admin settings and afterwards even override the admin settings again in the project settings. 
This is from my experience a good way to go if you have many projects to analyze where every project could be on a different Bitbucket server. (Example would be that customers have their own Bitbucket server)

onQualifiers(Qualifiers.APP) would remove the option to configure it per project and therefore remove this advantage of configuration

What's your point on that?

Reply:
Ok, I've got no issue with that. I just wanted to make sure you had a reason for using the one you'd chosen